rc: add numbered prefixes

SteVwonder opened the pull request:
- Add numbered prefixes to rc scripts
- Removes an unused sharness environment variable: FLUXION_RESOURCE_RC_PATH

Closes #587

dongahn commented on t/t0003-basic-install.t:
```diff
@@ -20,7 +20,7 @@ fi
 #
 SIZE=2
-if [ test ${FLUX_FLUX_CO_INST} = co ]; then
+if [ test ${FLUX_SCHED_CO_INST} != co ]; then
```
I think this will still fail, if a version of `flux-sched` was previously installed into `flux-core` you are testing this against...

If `FLUX_SCHED_CO_INST` == "co", this will use the installed RC scripts, which you probably want to avoid. 

I think this is the same issue as the problem you hit when you test PR #599...

SteVwonder replied:
Correct, this PR does not generically fix #607.  It enforces that only one set of flux-sched rc scripts runs, but unfortunately the ones it allows to run are the installed ones (not the in-tree ones).  I wasn't necessarily shooting to fix #607 with this PR.  Just patch up this test enough so that we have something to prevent a regression of the fix for #604.  We should definitely revisit this bit of logic once #607 is solved generically.

SteVwonder replied:
>  this will use the installed RC scripts, which you probably want to avoid.

Scratch my last response, as I think more about it, the whole intent of this script is to test the installed RC scripts and thus, by definition, ignore the rc scripts in-tree.  So we actually do want this behaviour for this test, but we don't always want this test to run (i.e., when your in-tree and install rc scripts differ).  We only want it to run when we intend to test the install scripts (i.e., in Travis CI), hence why this script is gated by the `FLUX_SCHED_TEST_INSTALLED` variable.  Sorry I didn't catch that earlier, it's getting late and I should probably step away from coding now :laughing:.

dongahn replied:
Assuming there will be a fix for ignoring installed third party RCs, this code appears correct to me.

dongahn commented on .travis.yml:
```diff
@@ -27,6 +27,7 @@ matrix:
       env:
        - ARGS="--prefix=/usr"
        - TEST_INSTALL=t
+       - FLUX_SCHED_TEST_INSTALLED=t
```
In my PR #599, I actually added this to `travis_run.sh` as it was exporting another variable. Perhaps doing it at one level down would be a bit cleaner?

SteVwonder replied:
Ah! Perfect!  I'll pull this line then, since your changes covers it.

SteVwonder commented:
Hmmm.  Got a failure just on Centos 7.  I'm tempted to say it's unrelated to this PR and might be a racy test.  The `stop due to alloc response error` below look ominous.  Going to try and restart the builder to see if it comes back.  Here are the relevant logs in case it doesn't reproduce.

test-script output:
```
expecting success: 

    flux module remove qmanager &&
    flux module load qmanager &&
    flux job list > jobs.list2 &&
    diff jobs.list2 jobs.list

2d1
< {"id": 11442061312, "userid": 2000, "priority": 16, "t_submit": 1582953004.6481304, "state": 4, "name": "hostname", "ntasks": 1, "t_depend": 1582953004.6481304, "t_sched": 1582953004.6600838}
not ok 5 - qmanager: restart keeps the main job queue intact
```

dmesg:
```
2020-02-29T05:10:04.619644Z qmanager.debug[0]: alloc success (id=10720641024)
2020-02-29T05:10:04.639460Z job-manager.debug[0]: submit_cb: added 1 jobs
2020-02-29T05:10:04.660067Z job-manager.debug[0]: submit_cb: added 1 jobs
2020-02-29T05:10:04.679174Z broker.debug[0]: rmmod qmanager
2020-02-29T05:10:04.679250Z job-manager.debug[0]: alloc: stop due to alloc response error: Function not implemented
2020-02-29T05:10:04.679278Z qmanager.debug[0]: service_unregister
2020-02-29T05:10:04.679342Z broker.debug[0]: module qmanager exited
2020-02-29T05:10:04.690664Z broker.debug[0]: insmod qmanager
2020-02-29T05:10:04.691011Z qmanager.debug[0]: service_register
2020-02-29T05:10:04.691084Z job-manager.debug[0]: scheduler: hello
2020-02-29T05:10:04.693787Z job-manager.debug[0]: scheduler: ready single
```

SteVwonder commented:
It didn't reproduce, so it looks like we have a race-condition bug on our hands.  I'll open an issue.

dongahn commented:
Probably the module got removed before the submitted job finished and we need to utilize some of the new sync primitives like `flux job wait`...

dongahn commented:
Once you remove your FLUX_SCHED_TEST_INSTALLED export commit and rebase, and travis CI passes, this looks good to me.

SteVwonder commented:
Just rebased and force-pushed, but it looks like the t0003-basic-install.t is not marked as failed when the broker encounters an error in rc3 scripts (i.e., when I revert 5191f03).  Lemme poke at this some to see if I can get the test to fail when that happens before we merge, otherwise it isn't really all that useful.

SteVwonder commented:
With the addition of b9ca320, `t0003-basic-install.t` will fail if 5191f03 is reverted, but it has the nasty side-effect of running every script in the installed rc3.d directory.  If flux-core puts any scripts there in the future, it will run them as well.  Not sure if it is the right solution.  Maybe the fix in #607 will help?

dongahn commented:
@SteVwonder: this is getting long in the tooth. Do we want to pick up something from this PR at this point?

SteVwonder commented:
Just rebased onto the latest master, dropped the added test which I believe we determined in no longer needed, and a commit that was superseded by other PRs.  Leaving the rc numbering change and the deletion of an obsolete variable.  Relatively small PR.

SteVwonder commented:
@dongahn, the numbered rc scripts has been added to #694, correct? If so, I think we can close this PR.

dongahn commented:
Ah yes. Closing this makes sense
